modal/app.py

```diff
@@ -869,7 +869,9 @@ def cls(
         scaledown_window: Optional[int] = None,  # Max time (in seconds) a container can remain idle while scaling down.
         proxy: Optional[_Proxy] = None,  # Reference to a Modal Proxy to use in front of this function.
         retries: Optional[Union[int, Retries]] = None,  # Number of times to retry each input in case of failure.
-        timeout: int = 300,  # Maximum execution time of the function in seconds.
+        # Maximum execution time of the function in seconds. This timeout applies independently to `@modal.enter` and
+        # functions.
```

Comment:
We should keep the documentation synced between `app.function()` and `app.cls()` IMO. (Of course the need to do this is part of why I don't like Cls, but still...

Also I think these inline parameter comments are a lot easier to read if they're all on the right, so unless you really need some verbose explanation, it's user-friendly to play with the language until you can make it fit. e.g.

> Maximum execution time of the function in seconds; applies independently to the startup phase and each input